docs/pages/guides/schema-validation.mdx

```diff
@@ -0,0 +1,124 @@
+
+The issue is that Storage with Liveblocks contains a different type structure than your application (client)- which could break your application.   How do you avoid this situation?  By enforcing Schema Validation.  While optional, Schema Validation is an extremely important tool for productionalized applications- it ensures that errors like the one described above could be avoided and resolved while working in development mode.  We'll elaborate a bit more on the inner workings of schema validation below.
+
+## How Schema Validation works
+As described above, the main purpose of Schema Validation is to prevent corrupted data from being loaded from the server. To accomplish this, two things need to happen. First, we need to ensure that data loaded into a Room matches the underlying Storage Tree and defined schema (if it exists). Second, we will introduce the rule that each operation or "Op" that occurs in the room which changes the Storage tree, will either be accepted or rejected.  
+
+When a Room is first loaded, the key `liveblocksSchema` from Storage is inspected: if no value is present, then schema validation is not enforced.  If there is a value, then that value of the schema is parsed and assigned to the Room where its values are distributed to the node tree.
+
+<!-- Insert an animation here !-->
+
+If schema validation is not enforced, all operations in the room are accepted. 
+For example, imagine that a client performs an operation that updates the Storage Tree:  regardless of whether or not this could break the application, Liveblocks has no opinion on whether or not this operation should be completed and accepts the incoming operation.  After the operation is completed, the updated Storage Tree is forwarded to other users in the room. 
+
+In the case where schema validation is enforced, when the client performs an operation that updates the Storage Tree, it is sent to the Liveblocks server which verifies that the updated Storage Tree is consistent with the defined schema attached to the Room.  If the verification succeeds, the Op is acknowledged by the initial client, and the Storage Tree is updated and forwarded to other users (clients) in the room.  If the operation verification fails, a `RejectedOpsServerMsg` is sent to the client.  The Storage Tree is not updated. In this instance, it is the responsibility of the developer to resolve the error.
+
+## Create a schema
+To create a schema, you will need to POST to the schemas endpoint:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+{
+  "id": "some-id",
+  "name": "Your schema name",
+  "data": `{
+    // your data schema will go here
+  }`,
+}
+
+type Body = {
+  id: string, // MAX_LENGTH = 64 / allowed characters az-
+  name: string, // MAX_LENGTH = 256
+  data: string,
+}
+```
+Note that the data type will be of type string because the schema will be string interpolated. 
```

Comment:
```suggestion
Note that the data type will be of type string because the schema will be parsed. For more information, see our [Schema Language documentation](xxx).
```